### PR TITLE
Expose the gwt.coverage flag in the gwt-maven-plugin

### PR DESCRIPTION
**Pull request by freongrr**
I want to generate code coverage reports for a GWT project. The GWT compiler can be passed a list of files it will instrument, but this is not convenient in my case. I'd rather be able to give a list of patterns and turn on/off instrumentation using maven profiles.

Typical usage:

    <plugin>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>gwt-maven-plugin</artifactId>
        <configuration>
            <coverage>
                <enabled>true</enabled>
                <scanDependencies>false</scanDependencies>
                <includes>
                    <include>com/myproject/client/**/*.java</include>
                </includes>
                <excludes>
                    <exclude>**/notneeded/**</exclude>
                </excludes>
            </coverage>
        </configuration>
    </plugin>

Please note that I'm rather new to Maven plugins so this may not be perfect (although it works nicely for me).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
     <version>30</version>
   </parent>
   <artifactId>gwt-maven-plugin</artifactId>
-  <version>2.7.0-SNAPSHOT</version>
+  <version>2.7.0-coverage-SNAPSHOT</version>
```
Comment (freongrr) [inline pom.xml]: Sorry, I did not see your comment earlier!
I changed the artifact id on my branch to avoid conflict while testing.

**Comment by ibaca** [comment]
Closing stalled pull requests. This might be useful, so if you really think this should be added please comment back. I have never used this coverage option, but… is not possible to get a similar result using just chrome devtools + source maps, or using V8/D8 from command line? Even if this is useful, adding just a plain command is much more clean and easier, is this filtering really needed?

**Comment by freongrr** [comment]
The coverage option of the compiler works well with automated tests. I have a suite of Selenium Webdriver tests that can run on Chrome, Firefox and IE. They extract the coverage data and build HTML reports using the Java source code.

Using the source maps would give me the Java code, but I would still need an automated way to extract the coverage data (potentially for multiple browsers).

I've rebased my changes on top of your master for GWT 2.8. I can create a new PR in case you're still interested: https://github.com/freongrr/gwt-maven-plugin

Also, what do you mean by "a plain command"?

**Comment by ibaca** [comment]
Hi, please create a new pull request so I remember that this is an open discussion. The nested configuration don't seems right, but the whole idea looks good. Also, please try to integrate this idea in the new plugin which should be ahead of this one. I'll try to review more seriously this weekend,  thanks!

**Comment by freongrr** [comment]
I see. I could move the configuration and generation of the `gwt-coverage.in` file to a different goal or plugin. I will still have to modify `CompileMojo` to pass the `gwt.coverage` arguments to the compiler though.

I've created a new PR ( #145).

**Pull request closed without merging**